modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java

```diff
@@ -0,0 +1,96 @@
+import org.testng.annotations.BeforeClass;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationBaseTest;
+import org.wso2.am.integration.test.utils.generic.APIMTestCaseUtils;
+import org.wso2.am.integration.test.utils.monitor.utils.WireMonitorServer;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.carbon.integration.common.admin.client.AuthenticatorClient;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * When we set "http.headers.preserve=Content-Type" in "passthru-http.properties" file,
+ * the charset part of the content type should be preserved for GET requests
+ * to address Endpoints with soap11 format.
+ */
+public class ESBJAVA3447PreserveCharsetInContentTypeTestCase extends APIMIntegrationBaseTest {
+
+    private static final Log log = LogFactory.getLog(ESBJAVA3447PreserveCharsetInContentTypeTestCase.class);
+
+    private WireMonitorServer wireServer;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+
+        wireServer = new WireMonitorServer(8992);
+
+        AuthenticatorClient login = new AuthenticatorClient(gatewayContextMgt.getContextUrls().getBackEndUrl());
+        String session = login.login("admin", "admin", "localhost");
+        // Upload the synapse
+        String file = "artifacts" + File.separator + "AM" + File.separator + "synapseconfigs" + File.separator
+                + "property" + File.separator + "CONTENT_TYPE_TEST.xml";
+        OMElement synapseConfig = APIMTestCaseUtils.loadResource(file);
+        APIMTestCaseUtils.updateSynapseConfiguration(synapseConfig, gatewayContextMgt.getContextUrls().getBackEndUrl(),
+                session);
+        Thread.sleep(5000);
+    }
+
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.ALL})
+    @org.testng.annotations.Test(groups = "wso2.am",
+            description = "Test for preserving Charset in Content-Type header in the request")
+    public void testPreserveCharsetInContentTypeHeader() throws Exception {
+
+        wireServer.start();
+        HttpClient httpclient = new DefaultHttpClient();
+        HttpGet httpget = new HttpGet(gatewayUrlsWrk.getWebAppURLNhttp() + "charset");
+        httpget.setHeader("Content-type", "text/xml; charset=UTF-8");
+        try {
+            httpclient.execute(httpget);
+        } catch (IOException e) {
+            log.error("Error in executing GET request for CharsetAPI",e);
+        }
+        String[] wireResponse = wireServer.getCapturedMessage().split(System.lineSeparator());
+        for (String line : wireResponse) {
+            if (line.contains("Content-Type")) {
+                if (!line.contains("text/xml; charset=UTF-8")) {
+                    Assert.fail("Content-Type header has dropped Charset");
+                }
+            }
+            Assert.assertTrue(true);
```

Comment:
This is added to pass the test when the content-Type header has not dropped the charset.